content/en/docs/controller-reference/replication-controller/component-subscription.md

```diff
@@ -0,0 +1,238 @@
+  semver: ">=6.3.x"
+  source:
+    url: ghcr.io/phoban01
+  destination:
+    url: ghcr.io/phoban01/foo
+    secretRef:
+      name: ghcr-credentials
+  verify:
+  - name: operations
+    publicKey:
+      secretRef:
+        name: signing-keys
+  - name: security
+    publicKey:
+      secretRef:
+        name: signing-keys
+```
+
+The accompanying secret should be in the following format:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: signing-keys
+type: Opaque
+data:
+  operations: <BASE64>
+  security: <BASE64>
+```
+
+## Debugging ComponentSubscriptions
+
+There are several ways to gather information about a ComponentSubscription for debugging purposes.
+
+### Describe the ComponentSubscription
+
+Describing an ComponentSubscription using `kubectl describe componentsubscription <subscription-name>` displays the latest recorded information for the resource in the Status sections:
+
+```bash
+...
+Status:
+  Conditions:
+    Last Transition Time:     2023-07-12T10:12:14Z
+    Message:                  no matching versions found for constraint '>=7.3.x'
+    Observed Generation:      2
+    Reason:                   PullingLatestVersionFailed
+    Status:                   False
+    Type:                     Ready
+  Last Applied Version:       6.3.6
+  Observed Generation:        1
+  Replicated Repository URL:  ghcr.io/phoban01/foo
+```
+
+Reconciliation errors are also logged by the controller. You can use a tool such as [stern](https://github.com/stern/stern) in tandem with `grep` to filter and refine the output of controller logs:
+
+```bash
+stern replication-controller -n ocm-system | grep ComponentSubscription
+```
+
+will output the following log stream:
+
+```bash
+replication-controller-76848b97c5-4flrl manager 2023-07-12T10:13:05Z    LEVEL(-4)       credentials configured  {"controller": "componentsubscription", "controllerGroup": "delivery.ocm.software", "controllerKind": "ComponentSubscription", "ComponentSubscription": {"name":"podinfo","namespace":"default"}, "namespace": "default", "name": "podinfo", "reconcileID": "a9eeba17-a533-4dc7-81fd-af97096d60aa"}
+replication-controller-76848b97c5-4flrl manager 2023-07-12T10:13:06Z    ERROR   Reconciler error        {"controller": "componentsubscription", "controllerGroup": "delivery.ocm.software", "controllerKind": "ComponentSubscription", "ComponentSubscription": {"name":"podinfo","namespace":"default"}, "namespace": "default", "name": "podinfo", "reconcileID": "a9eeba17-a533-4dc7-81fd-af97096d60aa", "error": "failed to get latest component version: no matching versions found for constraint '>=7.3.x'"}
+```
+
+## ComponentSubscription Status
+
+### Observed Generation
+
+The replication-controller reports an observed generation in the ComponentSubscription’s `.status.observedGeneration`. The observed generation is the latest `.metadata.generation` which resulted in either a ready state, or stalled due to error it can not recover from without human intervention.
```

Comment:
```suggestion
The replication-controller reports an observed generation in the ComponentSubscription's `.status.observedGeneration`. The observed generation is the latest `.metadata.generation`, which resulted in either a ready state or stalled due to an error it can not recover from without human intervention.
```